### DISCUSSION TITLE
404 on dynamic page only on Vercel, not in prod mode localhost

### DISCUSSION THREAD

**Discussion by JacoboArcila** [vercel/community#3775] [Help]
### Summary

I am using Pages Routes, when I deploy in Vercel my dynamic route /movies/[id].jsx throws a 404 error 
This is my function: 
```
export async function getServerSideProps(contexto) {
    const { id } = contexto.params;
    const movies = await getMovie(id);
    const cardMovies = await getMovies();

    return {
        props: {
            movies,
            cardMovies,
        },
    };
} 
```

I already contacted one of Next's support and he told me that it is probably a Vercel error

### Example

https://ssr-six-beta.vercel.app/

### Steps to Reproduce

To produce the 404 error, they have to scroll down and they have to click on see more in one of the cards that are seen.

When I make the page SSR, the 404 appears, but if I put it STATIC it works.
![image](https://github.com/vercel/community/assets/49046213/98d4f3d7-72a5-4a9d-920e-648c8942a35c)

These are the getMovies and getMovie functions:
![image](https://github.com/vercel/community/assets/49046213/d70d609c-aef0-4c38-8170-6cd99f56a306)

**Comment by amyegan** [comment]
Hey @JacoboArcila. If you share your public repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that will give me a better view so I can help debug.

If you can share a link to your previous Next.js discussion, that will also give me some context about what you already tried.

**Reply by JacoboArcila** [reply]
Hi @amyegan, this is the code link: https://github.com/JacoboArcila/SSR and the Vercel link: https://ssr-git-main-jacoboarcila.vercel.app/

And about the above discussion I don't have the link because it was on Discord, but I can take a screenshot.
![image](https://github.com/vercel/community/assets/49046213/8f38de2a-5f82-49ba-b6a4-ad18f50043f6)
